File: java64/nivel4/exer31.java
import javax.swing.JOptionPane;

public class exer31 {
    public static void main(String[] args) {
        int quantidadePar = 0, quantidadeImpar = 0, numero = 0, somaPar = 0, somaImpar = 0;
        
        while (numero >= 0) {
            
            numero = Integer.parseInt(JOptionPane.showInputDialog(null, "Informe o valor (números negativos finalizam): "));
            
            if (numero >= 0) {
                if (numero % 2 == 0) {
                    quantidadePar++;
                    somaPar += numero;
                    JOptionPane.showMessageDialog(null, "Par");
                } else {
                    quantidadeImpar++;
                    somaImpar += numero;
                    JOptionPane.showMessageDialog(null, "Impar");
                }
            }
        }
        JOptionPane.showMessageDialog(null, "Quantidade de números pares: " + quantidadePar
                + "\nQuantidade de números impares: " + quantidadeImpar);
        JOptionPane.showMessageDialog(null, "Soma dos números pares: " + somaPar
                + "\nSoma dos números impares: " + somaImpar);
    }
}

File: pyhton64/nivel4/exer31.py
numero = 0; qt_par = 0; soma_par = 0; qt_impar = 0; soma_impar = 0

while numero >= 0:
    numero = int(input('Informe o número: '))

    if numero >= 0:
        if numero % 2 == 0:
            qt_par += 1
            soma_par += numero
            print('Par')
        if numero % 2 == 1:
            qt_impar += 1
            soma_impar += numero
            print('Impar')
    else:
        break

print(f'\nPares: {qt_par}\nImpares: {qt_impar}')
print(f'\nSoma dos pares: {soma_par}\nSoma dos impares: {soma_impar}')

